pandas/src/algos_rank_helper.pxi.in

```diff
@@ -0,0 +1,392 @@
+"""
+Template for each `dtype` helper function for rank
+
+WARNING: DO NOT edit .pxi FILE directly, .pxi is generated from .pxi.in
+"""
+
+#----------------------------------------------------------------------
+# rank_1d, rank_2d
+#----------------------------------------------------------------------
+
+{{py:
+
+# name ctype pos_nan_value neg_nan_value
+dtypes = [('generic', 'object', 'Infinity()', 'NegInfinity()'),
```

Comment:
here I mean, I don't think we use generic generically (and instead use object) nomenclature